lib/toolbox/saved_struct.c

```diff
@@ -0,0 +1,149 @@
+#include "saved_struct.h"
+#include <furi.h>
+#include <stdint.h>
+#include <storage/storage.h>
+
+#define SAVED_STRUCT_TAG    "SAVED_STRUCT"
+
+typedef struct {
+    uint8_t magic;
+    uint8_t version;
+    uint8_t checksum;
+    uint8_t flags;
+    uint32_t timestamp;
+} SavedStructHeader;
+
+bool saved_struct_save(const char* path,
+                       void* data,
+                       size_t size,
+                       uint8_t magic,
+                       uint8_t version) {
+    furi_assert(path);
+    furi_assert(data);
+    furi_assert(size);
+    SavedStructHeader header;
+
+    FURI_LOG_I(SAVED_STRUCT_TAG, "Saving \"%s\"", path);
+
+    // Store
+    Storage* storage = furi_record_open("storage");
+    File* file = storage_file_alloc(storage);
+    bool result = true;
+    bool saved = storage_file_open(file, path, FSAM_WRITE, FSOM_CREATE_ALWAYS);
+    if(!saved) {
+        FURI_LOG_E(
+            SAVED_STRUCT_TAG,
+            "Open failed \"%s\". Error: \'%s\'",
+            path,
+            storage_file_get_error_desc(file));
+        result = false;
+    }
+
+    if(result) {
+        // Calculate checksum
+        uint8_t checksum = 0;
+        uint8_t* source = data;
```

Comment:
Doesn't make much sense. You already have unit8_t* to iterate with

Reply:
data is void*, to allow warning-less casting from other struct pointers.